src/storage.cpp

```diff
@@ -1055,7 +1057,7 @@ namespace libtorrent
 					}
 					if (ec)
 					{
-						ec.file = i->second;
+                        ec.file = i;
 						ec.operation = storage_error::copy;
 					}
 					else
```

Comment:
on line 1067, the whole directory structure is deleted. you may want to do something more clever there too. presumably stop deleting if a directory isn't empty, but delete each individual file that's copied